### PR TITLE
backport of my branches into upstream

### PR DESCRIPTION
**Pull request by pkit**
- new V8 version
- user context support
- `eval()` size limits support
- running script support
- various small fixes

### CODE REVIEW

File: plv8.cc [hunk 1]
```diff
@@ -260,96 +297,236 @@ CreateIsolate(plv8_context *context) {
...
-							256, 256, 3096, // hardcoded v8 limits for isolates
-							PGC_SUSET, 0,
+#define MAX_EVAL_SIZE_VAR "plv8.max_eval_size"
+	guc_value = plv8_find_option(MAX_EVAL_SIZE_VAR);
+	if (guc_value != NULL) {
+		plv8_max_eval_size = plv8_int_option(guc_value);
+	} else {
+		DefineCustomIntVariable(MAX_EVAL_SIZE_VAR,
+								gettext_noop("Max size in bytes for code to eval()"),
+								gettext_noop("The default is 2MB, setting to 0 disables eval() completely!"
```
Comment (pkit) [inline plv8.cc]: Probably need to change it to -1 by default for upstream to maintain existing behavior...

**Comment by pkit** [comment]
probably needs documentation updates too

**Comment by JerrySievert** [comment]
awesome! will take a look as soon as I get a free moment.  thanks @pkit!

**Comment by pkit** [comment]
Added some docs, may need to add more...

File: plv8.cc (hunk 1 above)
Comment (JerrySievert) [reply plv8.cc]: yes please, if you could in the PR.

**Comment by JerrySievert** [comment]
except for the one change, I'm happy to merge as-is.

I have some features I need to add for a 3.1 release, and adding those should do a good job of giving me a baseline against for native jsonb objects (in and out of plv8), adding tracing and more observability.

File: plv8.cc (hunk 1 above)
Comment (pkit) [reply plv8.cc]: Done

**Comment by JerrySievert** [comment]
awesome, thanks much!

**Pull request merged**